common/src/main/java/com/microsoft/identity/common/crypto/CryptoParameterSpecFactory.kt

```diff
@@ -0,0 +1,203 @@
+            digestAlgorithms = listOf(
+                KeyProperties.DIGEST_SHA256,
+                KeyProperties.DIGEST_SHA512
+            ),
+            description = MODERN_SPEC_WITH_PURPOSE_WRAP_KEY,
+            encryptionPaddings = getEncryptionPaddingsForKeyGen(),
+            algorithm = RSA_ALGORITHM
+        )
+    }
+
+    @delegate:RequiresApi(Build.VERSION_CODES.M)
+    private val keyGenParamSpecWithoutPurposeWrapKey by lazy {
+        KeyGenSpec(
+            keyAlias = keyAlias,
+            purposes = KeyProperties.PURPOSE_ENCRYPT or
+                    KeyProperties.PURPOSE_DECRYPT,
+            keySize = KEY_SIZE,
+            digestAlgorithms = listOf(
+                KeyProperties.DIGEST_SHA256,
+                KeyProperties.DIGEST_SHA512
+            ),
+            description = MODERN_SPEC_WITHOUT_PURPOSE_WRAP_KEY,
+            encryptionPaddings = getEncryptionPaddingsForKeyGen(),
+            algorithm = RSA_ALGORITHM
+        )
+    }
+
+    private val keyGenParamSpecLegacy = LegacyKeyGenSpec(
+        context = context,
+        keyAlias = keyAlias,
+        keySize = KEY_SIZE,
+        description = LEGACY_SPEC,
+        encryptionPaddings = listOf("PKCS1Padding"),
+        algorithm = RSA_ALGORITHM
+    )
+
+    @RequiresApi(Build.VERSION_CODES.M)
+    private fun getEncryptionPaddingsForKeyGen(): List<String> {
+        val paddings = mutableListOf(KeyProperties.ENCRYPTION_PADDING_RSA_PKCS1)
+        if (enableKeyGenEncryptionPaddingRsaOaep) {
+            paddings.add(KeyProperties.ENCRYPTION_PADDING_RSA_OAEP)
+        }
+        return paddings
+    }
+
+    /**
+     * Returns cipher specifications ordered by security preference.
+     *
+     * The list prioritizes OAEP padding (when enabled by feature flag) over PKCS1
+     * for better security, with PKCS1 always included for compatibility.
+     *
+     * @return List of [CipherSpec] objects ordered by preference (most secure first)
+     */
+    fun getPrioritizedCipherParameterSpecs(): List<CipherSpec> {
+        val methodTag = "$TAG:getPrioritizedCipherParameterSpecs"
+        val specs = listOf(
+            CipherSpec.oaepCipherSpec,
+            CipherSpec.pkcs1CipherSpec
+        )
+        Logger.info(methodTag, "Ciphers: $specs")
+        return specs
+    }
+
+    /**
+     * Returns key generation specifications ordered by API level compatibility.
+     *
+     * Prioritizes modern Android KeyStore features (API 23+) when enabled by feature flags,
+     * with legacy fallback always included for maximum compatibility.
+     *
+     * @return List of [IKeyGenSpec] objects ordered by preference (most modern first)
+     */
+    fun getPrioritizedKeyGenParameterSpecs(): List<IKeyGenSpec> {
+        val methodTag = "$TAG:getPrioritizedKeyGenParameterSpecs"
+        val specs = mutableListOf<IKeyGenSpec>()
+
+        // Add specs in order of preference
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.P && keySpecWithPurposeKey) {
```

Comment:
>Build.VERSION.SDK_INT >= Build.VERSION_CODES.P && keySpecWithPurposeKey [](http://example.com/codeflow?start=12&length=71)

can be in get()'s implemetnation

Reply:
Could you explain this in more detail? I’m not following.

Reply:
get() for this property can implementation like    

private val keySpecWithWrapPurposeKey get() = Build.VERSION.SDK_INT >= Build.VERSION_CODES.P && 
        flightsProvider.isFlightEnabled(CommonFlight.ENABLE_NEW_KEY_GEN_SPEC_FOR_WRAP_WITH_PURPOSE_WRAP_KEY)